benchmark/httpapi.py

```diff
@@ -163,6 +187,56 @@ def delete(self, request, id):
         request.setResponseCode(NO_CONTENT)
         return self.backend.delete(id)
 
+    @app.route("/benchmark-results", methods=['GET'])
+    def query(self, request):
+        """
+        Query the previously stored benchmarking results.
+
+        Currently this method only supports filtering the results by the
+        branch name.
+        There is no support for paging of results, but a limit on the
+        number of the results to return may be specified.
+        The returned results are ordered by the timestamp in descending
+        order.
+
+        :param twisted.web.http.Request request: The request.
+        """
+        request.setHeader(b'content-type', b'application/json')
+        params = self._parse_query_args(request.args)
+        d = self.backend.query(**params)
+
+        def got_results(results):
+            result = {"version": self.version, "results": results}
+            return dumps(result)
+
+        d.addCallback(got_results)
+        return d
+
+    @staticmethod
+    def _parse_query_args(args):
```

Comment:
+1 to not worth time for this code. When I was reviewing I wasn't sure what level of quality we were aiming for. This seems plenty fine for its use.